modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/subscription/SubscriptionManagerImpl.kt

```diff
@@ -296,7 +305,9 @@ class SubscriptionManagerImpl @Inject constructor(
     override fun refreshPurchases() {
         if (!billingClient.isReady) return
 
-        updateOfferEligibilityIfPurchaseHistoryExists()
+        applicationScope.launch {
+            updateOfferEligibilityIfPurchaseHistoryExists()
```

Comment:
I'm not sure I understand. You want me to change `queryPurchasesAsync()` to use `launch` as well?

Reply:
Sorry, my comment doesn't make sense. What I was trying to say is that now that the `updateOfferEligibilityIfPurchaseHistoryExists` is executed using a launch, and the next part of the code also uses a launch, could it cause an issue as they both update the same variable, `hasOfferEligible`? Should they be processed sequentially?

<img width="1639" alt="Screenshot 2024-08-26 at 3 41 16 PM" src="https://github.com/user-attachments/assets/868144f9-8674-4a9b-88b9-c027009e9e63">

Reply:
It might be. I was wondering about it as well when fixing things for the library bump. But this was the original behavior. I didn't want to change it.